includes/transformer/class-post.php

```diff
@@ -131,7 +131,19 @@ protected function get_actor_object() {
 	 *
 	 * @return string The Posts ID.
 	 */
-	protected function get_id() {
+	public function get_id() {
+		$last_legacy_id = (int) \get_option( 'activitypub_last_post_with_permalink_as_id', 0 );
+		$post_id        = $this->wp_object->ID;
+		$post_status    = \get_post_meta( $post_id, 'activitypub_status', true );
+
+		if (
+			'federated' !== $post_status &&
+			$post_id > $last_legacy_id
+		) {
+			// generate URI based on comment ID
+			return \add_query_arg( 'p', $post_id, \trailingslashit( \home_url() ) );
```

Comment:
Don't know if it may be of importance: Some popular plugins like "The Events Calendar" use a `guid` that differs from that scheme, for instance `https://wp.lan/?post_type=tribe_events&#038;p=8".

Any ideas, why using the GUID could be better or worse than composing an URI based on the post ID? ...uuups... that phpcomment is wrong, its mentioning the comment ID.

Reply:
@Menrath Good point.

I checked my local test env and my live system and the GUID is sadly not really used consistently!

* I see the permalink and the `?p=` version for Posts (no idea why WordPress is mixing it up)
* For Attachments it is the path to the Attachment file in the file system
* For pages it is the same as for Posts, sometimes it is the permalink, some times it is `?p=`

And what's even worse: Because the GUID includes the full domain, I found some old pages that still uses my old domain in it, because the rename script seems to not have properly worked!

But even if you take aside the permalink and domain issue, because it might be a historical issue, the Attachments would break everything.

Another point is, that the GUID can be set by implementers (as we do with Followers).

Reply:
@Menrath so I would stick with the current composing!?!

Reply:
I agree! Put thanks for pointing the arguments out again here, I think we already discussed them in private, but it's good to share the knowledge! :)